### PR TITLE
Fix #71 - Update to OpenSSL 1.1.1l

### LINKED ISSUE #71: Update to OpenSSL 1.1.1l
https://www.openssl.org/news/openssl-1.1.1-notes.html

### PR DESCRIPTION
**Pull request by jumde**
fix #71 

## Description

1. Update to OpenSSL 1.1.1l
2. Cleanup docker containers and images after build
3. Run shellcheck linter for shell scripts

### CODE REVIEW

File: build_linux.sh [hunk 1]
```diff
@@ -14,6 +14,8 @@ docker build -t tor-brave -f Dockerfile-linux \
     ${1+"$@"} .
 docker run --name tor-brave -d tor-brave
```
Comment (wknapik) [inline build_linux.sh:15]:
```shell
docker run --rm --name tor-brave -d tor-brave
```

Comment (wknapik) [inline build_linux.sh:15]:
```shell
docker run --rm --name tor-brave -d tor-brave
```

File: build_linux.sh [hunk 2]
```diff
@@ -14,6 +14,8 @@ docker build -t tor-brave -f Dockerfile-linux \
...
 docker cp tor-brave:/tor-$TOR_VERSION/install/bin/tor tor-$TOR_VERSION-linux-brave-$BRAVE_TOR_VERSION
+docker kill tor-brave
```
Comment (wknapik) [inline build_linux.sh:17]:
```shell
docker stop -t 15 tor-brave || docker kill tor-brave
```
would probably be better.

If you do `docker run --rm` above, you also won't have to do `docker rm` here.

File: build_mingw.sh [hunk 3]
```diff
@@ -14,3 +14,5 @@ docker build -t tor-brave-mingw -f Dockerfile-mingw \
     ${1+"$@"} .
 docker run --name tor-brave-mingw -d tor-brave-mingw
 docker cp tor-brave-mingw:/tor-$TOR_VERSION/install/bin/tor.exe tor-$TOR_VERSION-win32-brave-$BRAVE_TOR_VERSION.exe
+docker kill tor-brave-mingw
```
Comment (wknapik) [inline build_mingw.sh:17]: See: comments for build_linux.sh.

**Review by wknapik — CHANGES_REQUESTED** [review]
Please change the `ENTRYPOINT` in [Dockefile-linux](https://github.com/brave/tor_build_scripts/blob/e370d22018daf5cc76647fb75c619647cb922c08/Dockerfile-linux#L79) and [Dockerfile-mingw](https://github.com/brave/tor_build_scripts/blob/e370d22018daf5cc76647fb75c619647cb922c08/Dockerfile-mingw#L87). Those are still invalid and will fill up the disk on the jenkins node with an infinitely repeating error message (`dash: 1: read: arg count`). Once this is done, this is fine to merge from my pov.

Some remarks for the future:
* You can use `docker create` instead of `docker run` and you won't have to sleep forever in the `ENTRYPOINT`/`CMD`.
* I'd suggest using container IDs instead of setting fixed container names (`docker create` and `docker run -d` print the ID before exiting). This is more flexible (allows for concurrent runs) and avoids naming conflicts. You can label images and containers with `--label` for cleanup purposes.
* I'd also suggest use [hadolint](https://github.com/hadolint/hadolint) to lint dockerfiles.
* Standard Dockerfile stuff - drop root privileges, use proper init, use the exec form in `ENTRYPOINT`/`CMD`, etc. Hadolint would likely point some of these out.

**Review by wknapik — APPROVED** [review]

**Review by diracdeltas — APPROVED** [review]

**Review by darkdh — APPROVED** [review]

**Pull request merged**